packages/dataviews/src/view-table.tsx

```diff
@@ -48,7 +61,48 @@ const {
 	DropdownMenuSeparatorV2: DropdownMenuSeparator,
 } = unlock( componentsPrivateApis );
 
-function WithDropDownMenuSeparators( { children } ) {
+interface HeaderMenuProps< Item extends AnyItem > {
+	field: NormalizedField< Item >;
+	view: ViewTableType;
+	onChangeView: ( view: ViewTableType ) => void;
+	onHide: ( field: NormalizedField< Item > ) => void;
+	setOpenedFilter: ( fieldId: string ) => void;
+}
+
+interface BulkSelectionCheckboxProps< Item extends AnyItem > {
+	selection: string[];
+	onSelectionChange: ( items: Item[] ) => void;
+	data: Item[];
+	actions: Action< Item >[];
+}
+
+interface TableRowProps< Item extends AnyItem > {
+	hasBulkActions: boolean;
+	item: Item;
+	actions: Action< Item >[];
+	id: string;
+	visibleFields: NormalizedField< Item >[];
+	primaryField?: NormalizedField< Item >;
+	selection: string[];
+	getItemId: ( item: Item ) => string;
+	onSelectionChange: ( items: Item[] ) => void;
+	data: Item[];
+}
+
+interface ViewTableProps< Item extends AnyItem > {
+	actions: Action< Item >[];
+	data: Item[];
+	fields: NormalizedField< Item >[];
+	getItemId: ( item: Item ) => string;
+	isLoading?: boolean;
```

Comment:
Should most of these props be part of the base ViewProps?

Reply:
You're right, we should definitely unify that type. I'll do that.